Sources/OVP/Provider/OVPMediaProvider.swift

```diff
@@ -329,7 +330,20 @@ public enum OVPMediaProviderError: PKError {
                     }
                     
                     let mediaEntry: PKMediaEntry = PKMediaEntry(baseEntry.id, sources: mediaSources, duration: baseEntry.duration)
-                    let metaDataItems = self.getMetadata(metadataList: metadataList, partnerId: partnerId, entryId: mediaEntry.id)
+                    var metaDataItems = self.getMetadata(metadataList: metadataList, partnerId: partnerId, entryId: mediaEntry.id)
+                    
+                    if let baseEntry = baseEntry as? OVPExternalMediaEntry {
+                        
+                        if let sources = sources,
+                           baseEntry.externalSourceType != "YouTube" && sources.isEmpty {
+                            PKLog.debug("Response is not containing entry info or playback data.")
+                            callback(nil, OVPMediaProviderError.invalidResponse)
+                            return
+                        }
+                        
+                        metaDataItems["externalSourceType"] = baseEntry.externalSourceType
+                        metaDataItems["referenceId"] = baseEntry.referenceId
+                    }
```

Comment:
We need to take care of the case that the baseEntry is **not** an OVPExternalMediaEntry and the sources came back empty.

Reply:
done